### PR TITLE
Add the IAccessibleTextSelectionContainer interface

### PR DESCRIPTION
**Pull request by OzancanKaratas**
### Link to issue number:
None
### Summary of the issue:
None
### Description of how this pull request fixes the issue:
@michaelDCurran commited in iaccessible2 repository. I updated it in the NVDA repository.
### Testing strategy:
This is a submodule change. Test may not be necessary.
### Known issues with pull request:
None
### Change log entries:
Please refer the pull request in [LinuxA11y/IAccessible2/#17](https://github.com/LinuxA11y/IAccessible2/pull/17).

### CODE REVIEW

**Comment by leonardder** [comment]
I guess it doesn't make that much sense to implement this as long as it is not used in NVDA"s code base.

**Comment by OzancanKaratas** [comment]
You're right, but it can be used in the future. @michaelDCurran, what do you think?

**Comment by OzancanKaratas** [comment]
@michaelDCurran, tests fail. What should I do?

**Comment by seanbudd** [comment]
@OzancanKaratas - don't worry about unrelated test failures, we can retrigger the build as needed. I've done that now.

**Comment by OzancanKaratas** [comment]
@seanbudd, thanks for your interest. I couldn't find the reason for the failure. Can you help me please?

**Comment by michaelDCurran** [comment]
Note that the build failed due to a bug in the IAccessible2 IDL.
I have addressed this in linuxA11y/IAccessible2#19

There is no immediate need to include IAccessibleTextSelectionContainer in NVDA, until Chrome or another web browser have implemented it.

**Comment by OzancanKaratas** [comment]
> There is no immediate need to include IAccessibleTextSelectionContainer in NVDA, until Chrome or another web browser have implemented it.

Yes, but I wanted to prepare NVDA for the future.

**Comment by seanbudd** [comment]
Keeping this as a draft until this is ready

**Comment by OzancanKaratas** [comment]
@seanbudd, I apologize for the inconveniences. It is now ready.

**Comment by lukaszgo1** [comment]
Given [this comment](https://github.com/nvaccess/nvda/pull/13400#issuecomment-1059899014) by @michaelDCurran  shouldn't this PR be closed as the new interface is not supported by any of the browsers currently.

**Comment by seanbudd** [comment]
@OzancanKaratas - what does merging this solve? Is there a summary of the changes introduced by changing commits?

**Comment by seanbudd** [comment]
Waiting on browsers to implement this

**Comment by seanbudd** [comment]
It appears that chromium has implemented this https://chromium-review.googlesource.com/c/chromium/src/+/3690092

File: nvdaHelper/ia2_sconscript [hunk 1]
```diff
@@ -44,11 +44,11 @@ idlFiles = [
 	'AccessibleStates.idl',
 	'Accessible2.idl',
 	'Accessible2_2.idl',
-	'Accessible2_3.idl',
```
[the remark this answers is not in the document]
Comment (OzancanKaratas) [reply nvdaHelper/ia2_sconscript:47]: Because this has been removed from the original repository.

[the remark this answers is not in the document]
Comment (seanbudd) [reply nvdaHelper/ia2_sconscript:47]: No it hasn't.
If you check the diff of this PR, that file has not been deleted in the submodule update.
The file is also on the master branch
https://github.com/LinuxA11y/IAccessible2/blob/master/api/Accessible2_3.idl

**Review by seanbudd — APPROVED** [review]
Thanks @OzancanKaratas, I am going to hold off on merging until we can discuss and create an issue to make use of this on NVDA's side, now that chromium supports this

Edit: this is being tracked internally, just going to await @michaelDCurran to confirm this is ok to merge.

**Comment by seanbudd** [comment]
Blocked by further work on the implementation by Chrome/Firefox

**Comment by seanbudd** [comment]
I'm closing this as invalid. There still needs to be work done on the chromium side

**Pull request closed without merging**